core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -374,28 +374,33 @@ class RequestMetrics(name: String) extends KafkaMetricsGroup {
     else
       None
 
-  private val errorMeters = mutable.Map[Errors, ErrorMeter]()
-  Errors.values.foreach(error => errorMeters.put(error, new ErrorMeter(name, error)))
+  private val errorMeters = Errors.values.map(error => (error, new ErrorMeter(name, error))).toMap
 
   class ErrorMeter(name: String, error: Errors) {
     private val tags = Map("request" -> name, "error" -> error.name)
 
     @volatile private var meter: Meter = null
+    private var deleted: Boolean = false
```

Comment:
Hmm, is there a benefit in tracking this per meter instead of having a `closed` global flag?

Reply:
It just felt simpler with a flag in the `ErrorMeter` since we already synchronize on this for creating meters. A global one would need different locking.

Reply:
What if we changed `markErrorMeter` to call `errorMeters.get` instead of `errorMeters.apply`? Would that not be sufficient?

Reply:
Hmm.. Is it sufficient? A thread could be in the middle of creating and we may have a meter that is not deleted.